File: packages/phaser-cv-demo/scripts/capture-cv-output.py
#!/usr/bin/env python3
"""Connect to CV WebSocket, wait for one frame_processed, print detections.
Run while game is open at ?cv=1 - press S in game to trigger capture."""
import asyncio
import json
import sys

sys.path.insert(0, str(__import__('pathlib').Path(__file__).resolve().parent.parent.parent / 'phaser-cv' / 'cv-bridge'))
try:
    import websockets
except ImportError:
    print("Install: pip install websockets")
    sys.exit(1)

async def main():
    print("[capture-cv] Connecting...")
    async with websockets.connect("ws://localhost:8765") as ws:
        print("[capture-cv] Connected. Waiting for frame... (press S in game)")
        async for raw in ws:
            msg = json.loads(raw)
            if msg.get("type") == "frame_processed":
                print("\n=== CV DETECTIONS (full) ===\n")
                print(json.dumps(msg.get("detections", {}), indent=2))
                print("\n=== END ===\n")
                return

asyncio.run(asyncio.wait_for(main(), timeout=15))

File: packages/phaser-cv-demo/scripts/capture-cv-output.mjs
#!/usr/bin/env node
/**
 * Connect to CV WebSocket, wait for one frame_processed broadcast, print detections.
 * Run while game is open at ?cv=1 - press S in game to trigger capture.
 */
import { createRequire } from 'module';
const require = createRequire(import.meta.url);
const WebSocket = require('ws');

const ws = new WebSocket('ws://localhost:8765');
ws.on('open', () => console.log('[capture-cv] Connected, waiting for frame... (press S in game)'));
ws.on('message', (data) => {
  try {
    const msg = JSON.parse(data.toString());
    if (msg.type === 'frame_processed') {
      console.log('\n=== CV DETECTIONS (full) ===\n');
      console.log(JSON.stringify(msg.detections, null, 2));
      console.log('\n=== END ===\n');
      ws.close();
      process.exit(0);
    }
  } catch (e) {
    console.error(e);
  }
});
ws.on('error', (e) => {
  console.error('[capture-cv] Error:', e.message);
  process.exit(1);
});
setTimeout(() => {
  console.error('[capture-cv] Timeout 15s - no frame received. Is game open at ?cv=1? Press S?');
  process.exit(1);
}, 15000);
